libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp

```diff
@@ -477,11 +496,21 @@ void AP_RangeFinder_LightWareI2C::legacy_timer(void)
 
 void AP_RangeFinder_LightWareI2C::sf20_timer(void)
 {
-    if (sf20_get_reading(state.distance_cm)) {
-        // update range_valid state based on distance measured
-        update_status();
+    // If no read errors, try to perform nominal readings
+    if (read_errors_ < MAX_READ_ERRORS_THRESHOLD) {
+        if (sf20_get_reading(state.distance_cm)) {
+            // Update range_valid state based on distance measured
+            update_status();
+        } else {
+            set_status(RangeFinder::RangeFinder_NoData);
+        }
     } else {
-        set_status(RangeFinder::RangeFinder_NoData);
+        // Otherwise try to init again.
```

Comment:
Could you add `set_status(RangeFinder::RangeFinder_NoData);` for this else case?

Reply:
Sure. Done!